docs/EN_US/ECLStandardLibraryReference/SLR-Mods/DataPatternsLibrary.xml

```diff
@@ -15,59 +15,708 @@
+      </indexterm><indexterm>
+        <primary>DataPatterns.BestRecordStructure</primary>
+      </indexterm><indexterm>
+        <primary>BestRecordStructure</primary>
+      </indexterm></emphasis></para>
+
+    <para>Function macro that leverages DataPatterns to return a string
+    defining the best ECL record structure for the input data.</para>
+
+    <informaltable colsep="1" frame="all" rowsep="1">
+      <tgroup cols="2">
+        <tbody>
+          <row>
+            <entry><emphasis>inFile</emphasis></entry>
+
+            <entry>The dataset to process; REQUIRED</entry>
+          </row>
+
+          <row>
+            <entry><emphasis>sampling</emphasis></entry>
+
+            <entry>A positive integer representing a percentage of inFile to
+            examine, which is useful when analyzing a very large dataset and
+            only an estimatation is sufficient; valid range for this argument
+            is 1-100; values outside of this range will be clamped; OPTIONAL,
+            defaults to 100 (which indicates that the entire dataset will be
+            analyzed)</entry>
+          </row>
+
+          <row>
+            <entry><emphasis>emitTransform</emphasis></entry>
+
+            <entry>Boolean governing whether the function emits a TRANSFORM
+            function that could be used to rewrite the dataset into the 'best'
+            record definition; OPTIONAL, defaults to FALSE.</entry>
+          </row>
+
+          <row>
+            <entry><emphasis>textOutput</emphasis></entry>
+
+            <entry> Boolean governing the type of result that is delivered by
+            this function; if FALSE then a recordset of STRINGs will be
+            returned; if TRUE then a dataset with a single STRING field, with
+            the contents formatted for HTML, will be returned (this is the
+            ideal output if the intention is to copy the output from ECL
+            Watch); OPTIONAL, defaults to FALSE</entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </informaltable>
+
+    <para>The best record structure returns a recordset defining the best ECL
+    record structure for the data. If textOutput is FALSE (the default) then
+    each record will contain one field declaration, and the list of
+    declarations will be wrapped with RECORD and END strings; if the
+    emitTransform argument was TRUE, there will also be a set of records that
+    that comprise a stand-alone TRANSFORM function. If textOutput is TRUE then
+    only one record will be returned, containing an HTML-formatted string
+    containing the new field declarations (and optionally the TRANSFORM); this
+    is the ideal format if the intention is to copy the result from ECL
+    Watch.</para>
+
+    <para><emphasis role="bold">Example:</emphasis></para>
+
+    <programlisting>BestRecordStructure(inFile, sampling = 100, emitTransform = FALSE, textOutput = FALSE) := FUNCTIONMACRO
```

Comment:
This appears to be the definition of the function macro rather than an example programmatic invocation of profiling.  Is it useful, but an example call would also be very helpful.